src/ucp/core/ucp_am.c

```diff
@@ -910,6 +910,10 @@ ucp_am_send_req(ucp_request_t *req, size_t count,
      * Otherwise, return the request.
      */
     ucp_request_send(req);
+    if (ucs_unlikely(param == NULL)) {
```

Comment:
isn't it too late to check? Note that it previously used on lines 860 and 886

Reply:
it's not used for short/EAGER flow. I agree that it looks hacky but on other hand it could code duplication. Any idea how to refactor this code is welcome

Reply:
for instance using the empty params structure instead of NULL

Reply:
applied, pls take a look if it looks better for you

Reply:
but it seems that the empty params  is valid use case for `ucp_am_send_nbx` and may cause unexpected behavior from perspective of this PR